plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -774,7 +774,12 @@ protected ResultSet getTables(Connection connection, Optional<String> schemaName
                 connection.getCatalog(),
                 escapeNamePattern(schemaName, metadata.getSearchStringEscape()).orElse(null),
                 escapeNamePattern(tableName, metadata.getSearchStringEscape()).orElse(null),
-                new String[] {"TABLE", "VIEW"});
+                getTableTypes().toArray(String[]::new));
```

Comment:
Do we want to allow subclasses to provide `null` here (perhaps as `Optional<List<String>>`)?

Reply:
Good point.

Reply:
ac